### PR TITLE
Added documentation regarding external networks

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

File: docs/source/usage/common_scenarios.rst [hunk 1]
```diff
@@ -69,4 +69,52 @@ In order for the lessee to access a node on a private network, they can do one o
...
+
+Once that's done, you can create a floating IP and associate it with the node's port (which can be found by running ``openstack esi node network list``):
+
+  .. prompt:: bash $
+
+    openstack floating ip create <external network>
+    openstack floating ip set --port <port> <floating ip>
+
+
+Trunk Port
```
Comment (naved001) [inline docs/source/usage/common_scenarios.rst:98]: I think we should call this section something like "Connect a node directly to the external network VLAN". And then using trunk port is one of the scenarios.

1. If we have only a single NIC and we want both, the provisioning network *and* an external network, then we need a trunk port.
2. If we have multiple NICs, we can follow the instructions to attach the external network on the other interface and configure that. 
3. If we a single NIC and serial console access, we could replace the provisioning network with the external network (so need of trunk port in this case).

Comment (naved001) [reply docs/source/usage/common_scenarios.rst:98]: Also, just like other networks, an administrator must allow the project access to the external network for this to work. Right?

File: docs/source/usage/common_scenarios.rst [hunk 2]
```diff
@@ -69,4 +69,52 @@ In order for the lessee to access a node on a private network, they can do one o
...
     openstack esi node network attach --trunk <trunk name> <node>
 
+Using External Networks
+-----------------------
+
+The recommended way to allow external network access to a baremetal node is by `creating an external provider network`_. The instructions in the link explain how; in addition, this network should be created as ``--provider-network-type vlan`` with the appropriate vlan specified by ``--provider-segment``.
+
+Once that's done, a lessee can provision a node upon their provisioning network and then gain external network access using one of two methods.
+
+Floating IP
```
Comment (naved001) [inline docs/source/usage/common_scenarios.rst:79]: Do OpenStack security groups play a part here when we actually want to reach the node? In the VM world we have to explicitly enable what ports are allowed for the VM.

**Review by naved001 — COMMENTED** [review]
just a bunch of general questions.

File: docs/source/usage/common_scenarios.rst [hunk 3]
```diff
@@ -69,4 +69,52 @@ In order for the lessee to access a node on a private network, they can do one o
...
+    openstack router create external-router
+    openstack router set --external-gateway <external network> external-router
+    openstack router add subnet external-router <provisioning subnet>
+
+Once that's done, you can create a floating IP and associate it with the node's port (which can be found by running ``openstack esi node network list``):
+
+  .. prompt:: bash $
+
+    openstack floating ip create <external network>
+    openstack floating ip set --port <port> <floating ip>
```
Comment (DanNiESh) [inline docs/source/usage/common_scenarios.rst:95]: Does this step happen after the node is provisioned with provisioning network?

File: docs/source/usage/common_scenarios.rst (hunk 2 above)
Comment (tzumainn) [reply docs/source/usage/common_scenarios.rst:79]: From my testing, it actually does not.

File: docs/source/usage/common_scenarios.rst (hunk 3 above)
Comment (tzumainn) [reply docs/source/usage/common_scenarios.rst:95]: Yep!

File: docs/source/usage/common_scenarios.rst (hunk 1 above)
Comment (tzumainn) [reply docs/source/usage/common_scenarios.rst:98]: > I think we should call this section something like "Connect a node directly to the external network VLAN". And then using trunk port is one of the scenarios.
> 
> 1. If we have only a single NIC and we want both, the provisioning network _and_ an external network, then we need a trunk port.
> 2. If we have multiple NICs, we can follow the instructions to attach the external network on the other interface and configure that.
> 3. If we a single NIC and serial console access, we could replace the provisioning network with the external network (so need of trunk port in this case).

Fair enough! The only thing I haven't actually tested 2) or 3) so I'm reluctant to document that here for now. I'll add 1) though.

Comment (naved001) [reply docs/source/usage/common_scenarios.rst:98]: Makes sense.

**Comment by tzumainn** [comment]
I've made some changes - let me know if you guys think more are needed!

**Review by naved001 — APPROVED** [review]

**Review by DanNiESh — APPROVED** [review]
LGTM!

**Comment by tzumainn** [comment]
Thanks all!

**Pull request merged**